### DISCUSSION TITLE
Spectral environment emitter

### DISCUSSION THREAD

**Discussion by UNakade** [mitsuba-renderer/mitsuba3#183] [Q&A] [answered]
Is it possible to have an environment emitter with a spectral specification of radiance instead of an RGB image?

**Comment by njroussel** [comment]
Currently I do not believe that this is possible.
The `envmap` plugin will always try to upsample (rgb -> spectrum) the data it has read through the `filename` property when used with `_spectal` variant.

**Comment by Speierers** [comment, marked as answer]
Indeed this is not supported unfortunately. 

It isn't currently on our roadmap, but if I were to implement this, here if how I would do it:
- Implement a `SpectralBitmap` texture plugin that loads a "spectral image file" interpolates between the right channels given the queried wavelength.
- Implement a `SpectralEnvmap` emitter plugin that uses the `SpectralBitmap` plugin.

If you give this a try, please open a PR and I will be more than happy to help you with this.

**Reply by diiigle** [reply]
As a "spectral image file" you could use the [proposed OpenEXR layout](https://jcgt.org/published/0010/03/01/) of @afichet
Comes with [code](https://github.com/afichet/spectral-exr).
